charmcraft/commands/pack.py

```diff
@@ -153,8 +149,12 @@ def run(self, parsed_args: argparse.Namespace) -> None:
                     f"Currently trying to pack: {self.config.project.dirpath}"
                 )
             self._check_config(bases=True)
-            pack_method = self._pack_charm
+            with instrum.Timer("Whole pack run"):
+                self._pack_charm(parsed_args, builder)
         elif self.config.type == "bundle":
+            if parsed_args.shell:
+                package.launch_shell()
+                return
```

Comment:
Why is this defined only for bundles instead of having common shell handling for both types?